src/main/java/amazin/controller/BookController.java

```diff
@@ -107,11 +112,26 @@ public String updateBook(@PathVariable("id") Long id, @Valid @ModelAttribute(MOD
 
     @PostMapping("delete/{id}")
     public String deleteBook(@PathVariable("id") Long id) {
-        bookService.delete(id);
+        Optional<Book> deletedBook = bookService.delete(id);
+
+        if (deletedBook.isPresent()) {
+            Book book = deletedBook.get();
+            deleteFile(book.getPicture_url());
+        }
 
         return createRedirectViewPath(REQUEST_MAPPING_BOOK);
     }
 
+    @PostMapping("/storage/uploadFile")
+    public String uploadFile(@RequestPart(value = "file") MultipartFile file) {
+        return this.amazonService.uploadFile(file);
+    }
+
+    @DeleteMapping("/storage/deleteFile")
```

Comment:
We might want to later rename these to /storage/deletePicture and /storage/uploadPicture since we aren't really allowing users to upload other files.